tests/nominal/test_OneHotEncodingTransformer.py

```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
             actual=df_transformed,
             msg="unseen category rows not encoded as 0s",
         )
+
+
+class TestAdditionalOneHotEncodingTransformer:
+    """Additional tests for OneHotEncodingTransformer."""
+
+    @pytest.fixture
+    def sample_data(self):
+        return pd.DataFrame(
+            {
+                "color": ["red", "blue", "green", "blue", "red"],
+                "size": ["S", "M", "L", "S", "M"],
+            },
+        )
+
+    def test_not_fitted_error_transform(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        with pytest.raises(NotFittedError):
+            transformer.transform(sample_data)
+
+    def test_initialization_and_fitting(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        transformer.fit(sample_data)
+        assert hasattr(
+            transformer._encoder,
+            "categories_",
+        ), "The encoder should have 'categories_' attribute after fitting."
+
+    def test_transformation(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        transformer.fit(sample_data)  # Ensure the transformer is fitted
+        transformed = transformer.transform(sample_data)
+        expected_columns = [
+            "color_red",
+            "color_blue",
+            "color_green",
+            "size_S",
+            "size_M",
+            "size_L",
+        ]
+        assert all(
+            col in transformed.columns for col in expected_columns
+        ), "All expected columns should be in the transformed dataframe."
+
+    def test_unseen_categories(self, sample_data):
```

Comment:
again, duplication of tests in TestTransform

Reply:
remove